uint8   ctrl_mode
uint8   robot_mode
uint32  error
uint32  warning